File: Python/27. Remove Element.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        if not nums:
            return 0

        k = 1  # Count of unique elements
        slow = 0  # Pointer to track the last unique element

        for fast in range(1, len(nums)):
            if nums[fast] != nums[slow]:
                slow += 1
                nums[slow] = nums[fast]
                k += 1

        del nums[slow + 1 :]  # Remove the remaining duplicates from the list

        return k

File: Javascript/27. Remove Element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
	if (!nums) {
		return 0;
	}

	let k = 0;
	let slow = -1;

	for (let fast = 0; fast < nums.length; fast++) {
		if (nums[fast] !== val) {
			k += 1;
			slow += 1;
			nums[slow] = nums[fast]
		}

	}
	return k;
};